pkg/destroy/openstack/openstack.go

```diff
@@ -624,15 +624,14 @@ func deleteCustomRouterInterfaces(opts *clientconfig.ClientOpts, filter Filter,
 		return false, nil
 	}
 
-	// Find the machines Network by the master machines Ports.
+	// Find the ports that should be removed from the router.
 	// Any worker nodes ports, including the ones used for additionalNetworks,
 	// are tagged with cluster-api-provider-openstack and should be excluded
 	// to ensure only the primary Network ports are filtered.
 	tags := filterTags(filter)
 	listOpts := ports.ListOpts{
-		Tags:        strings.Join(tags, ","),
-		DeviceOwner: "compute:nova",
```

Comment:
Probably the reason this code didn't work on the CI is that the device-owner is `compute:central` instead of `compute:nova`